Firestore/tests/System/DocumentAndCollectionTest.php

```diff
@@ -47,7 +47,10 @@ public function testCreate()
 
     public function testSnapshotWithReadTime()
     {
-        $readTime = new Timestamp(new \DateTimeImmutable());
+        // read snapshot first
+        $snapshot = $this->document->snapshot();
+        // use "now" as last read time
```

Comment:
@bshaffer I think the real issue here maybe different. I ran the phpunit with `--repeat 10` flag and I can repro this test failures intermittently with `testSnapshotWithReadTime`. I believe the feature snapshot reads needs some time to show up the latest document. I can see that in cases when tests fail, a null document reference is returned and [SnapshotTrait's code pointer](https://github.com/googleapis/google-cloud-php/blob/b183f111255842d6ef4aadc51960ebe56c45e5f0/Firestore/src/SnapshotTrait.php#L61) returns a NotFoundException. So,

1. Per feature design, a document snapshot read shouldn't be required before reading at a timestamp.
2. `now` shouldn't be needed since PHP defines the default param as the same `DateTimeImmutable::__construct($datetime = 'now')`, [ref](https://www.php.net/manual/en/datetimeimmutable.construct.php).
3. If we add `sleep(1)` then the test starts passing consistently.

```suggestion
        // without sleep, test fails intermittently
        sleep(1);
```

Reply:
1. Okay, that's good to know
2. I just put it there for readability
3. That's great! I agree it's a better solution, and I verified locally that it works